{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6f9eff61",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip\n",
    "\n",
    "def get_vcf_names(vcf_path):\n",
    "    with gzip.open(vcf_path, \"rt\") as ifile:\n",
    "          for line in ifile:\n",
    "            if line.startswith(\"#CHROM\"):\n",
    "                  vcf_names = [x for x in line.split('\\t')]\n",
    "                  break\n",
    "    ifile.close()\n",
    "    return vcf_names\n",
    "\n",
    "\n",
    "names = get_vcf_names('/home/kasia/mnm/CPCT02220079.annotated.processed.vcf.gz')\n",
    "vcf = pd.read_csv('/home/kasia/mnm/CPCT02220079.annotated.processed.vcf.gz', compression='gzip', comment='#', chunksize=10000, delim_whitespace=True, header=None, names=names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "57dc9a08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#CHROM',\n",
       " 'POS',\n",
       " 'ID',\n",
       " 'REF',\n",
       " 'ALT',\n",
       " 'QUAL',\n",
       " 'FILTER',\n",
       " 'INFO',\n",
       " 'FORMAT',\n",
       " 'CPCT02220079R\\n']"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "37c9332f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam\n",
    "\n",
    "vcf_in = pysam.VariantFile(\"/home/kasia/mnm/CPCT02220079.annotated.processed.vcf.gz\")\n",
    "vcf_out = pysam.VariantFile(\"/home/kasia/mnm/CPCT02220079.annotated.processed_filtered.vcf.gz\", 'w', header=vcf_in.header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "11ec00e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "for rec in vcf_in.fetch('12',112,112):\n",
    "    vcf_out.write(rec)\n",
    "#for rec in vcf_in.fetch('12',204,247):\n",
    "#    vcf_out.write(rec)\n",
    "#for rec in vcf_in.fetch('12',691,789):\n",
    "#    vcf_out.write(rec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "b2d3f0d4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pysam.libcbcf.TabixIterator object at 0x7f24fec042c0>\n"
     ]
    }
   ],
   "source": [
    "print(vcf_in.fetch('1', 112, 112))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "91296843",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "\n",
      "GENERIC fileformat\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "\n",
      "FILTER FILTER\n",
      "##ALT=<ID=NON_REF,Description=\"Represents any possible alternative allele at this location\">\n",
      "\n",
      "STRUCTURED ALT\n",
      "##FILTER=<ID=INDEL_LowQualityDepth,Description=\"QD < 2.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=INDEL_ReadPosRankSumLow,Description=\"ReadPosRankSum < -20.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=INDEL_StrandBias,Description=\"FS > 200.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=LowQual,Description=\"Low quality\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SNP_HaplotypeScoreHigh,Description=\"HaplotypeScore > 13.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SNP_LowQualityDepth,Description=\"QD < 2.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SNP_MQRankSumLow,Description=\"MQRankSum < -12.5\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SNP_MappingQuality,Description=\"MQ < 40.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SNP_ReadPosRankSumLow,Description=\"ReadPosRankSum < -8.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SNP_StrandBias,Description=\"FS > 60.0\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FILTER=<ID=SnpCluster,Description=\"SNPs found in clusters\">\n",
      "\n",
      "FILTER FILTER\n",
      "##FORMAT=<ID=AD,Number=R,Type=Integer,Description=\"Allelic depths for the ref and alt alleles in the order listed\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=DP,Number=1,Type=Integer,Description=\"Approximate read depth (reads with MQ=255 or with bad mates are filtered)\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=GQ,Number=1,Type=Integer,Description=\"Genotype Quality\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=MIN_DP,Number=1,Type=Integer,Description=\"Minimum DP observed within the GVCF block\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##FORMAT=<ID=SB,Number=4,Type=Integer,Description=\"Per-sample component statistics which comprise the Fisher's Exact Test to detect strand bias.\">\n",
      "\n",
      "FORMAT FORMAT\n",
      "##GATKCommandLine.CombineVariants=<ID=CombineVariants,Version=3.8-0-ge9d806836,Date=\"Tue May 28 02:42:58 CEST 2019\",Epoch=1559004178928,CommandLineOptions=\"analysis_type=CombineVariants input_file=[] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=null excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/data/GENOMES/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=1000 baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=1 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false variant=[(RodBindingCollection [(RodBinding name=variant source=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_snps.vcf)]), (RodBindingCollection [(RodBinding name=variant2 source=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_indels.vcf)])] out=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_variants.vcf genotypemergeoption=null filteredrecordsmergetype=KEEP_IF_ANY_UNFILTERED multipleallelesmergetype=BY_TYPE rod_priority_list=null printComplexMerges=false filteredAreUncalled=false minimalVCF=false excludeNonVariants=false setKey=set assumeIdenticalSamples=true minimumN=1 suppressCommandLineHeader=false mergeInfoWithMaxAC=false filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false\">\n",
      "\n",
      "STRUCTURED GATKCommandLine.CombineVariants\n",
      "##GATKCommandLine.GenotypeGVCFs=<ID=GenotypeGVCFs,Version=3.8-0-ge9d806836,Date=\"Tue May 28 02:15:14 CEST 2019\",Epoch=1559002514021,CommandLineOptions=\"analysis_type=GenotypeGVCFs input_file=[] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=[/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/.queue/scatterGather/GermlineCaller-2-sg/temp_001_of_256/scatter.intervals] excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/data/GENOMES/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=1000 baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=2 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false variant=[(RodBindingCollection [(RodBinding name=variant source=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/CPCT02220079R_dedup.realigned.g.vcf.gz)])] out=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/.queue/scatterGather/GermlineCaller-2-sg/temp_001_of_256/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.raw_variants.vcf includeNonVariantSites=false uniquifySamples=false annotateNDA=false useNewAFCalculator=false heterozygosity=0.001 indel_heterozygosity=1.25E-4 heterozygosity_stdev=0.01 standard_min_confidence_threshold_for_calling=10.0 standard_min_confidence_threshold_for_emitting=30.0 max_alternate_alleles=6 max_genotype_count=1024 max_num_PL_values=100 input_prior=[] sample_ploidy=2 annotation=[] group=[StandardAnnotation] dbsnp=(RodBinding name=dbsnp source=/data/common_dbs/GATK_bundle_v2.8/dbsnp_137.b37.vcf.gz) filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false\">\n",
      "\n",
      "STRUCTURED GATKCommandLine.GenotypeGVCFs\n",
      "##GATKCommandLine.HaplotypeCaller=<ID=HaplotypeCaller,Version=3.8-0-ge9d806836,Date=\"Mon May 27 21:29:13 CEST 2019\",Epoch=1558985353803,CommandLineOptions=\"analysis_type=HaplotypeCaller input_file=[/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/CPCT02220079R/mapping/CPCT02220079R_dedup.realigned.bam] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=[/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/.queue/scatterGather/GermlineCaller-1-sg/temp_001_of_256/scatter.intervals] excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/data/GENOMES/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=500 baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=1 num_cpu_threads_per_data_thread=2 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=LINEAR variant_index_parameter=128000 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false likelihoodCalculationEngine=PairHMM heterogeneousKmerSizeResolution=COMBO_MIN dbsnp=(RodBinding name= source=UNBOUND) dontTrimActiveRegions=false maxDiscARExtension=25 maxGGAARExtension=300 paddingAroundIndels=150 paddingAroundSNPs=20 comp=[] annotation=[StrandBiasBySample] excludeAnnotation=[ChromosomeCounts, FisherStrand, StrandOddsRatio, QualByDepth] group=[StandardAnnotation, StandardHCAnnotation] debug=false useFilteredReadsForAnnotations=false emitRefConfidence=GVCF bamOutput=null bamWriterType=CALLED_HAPLOTYPES emitDroppedReads=false disableOptimizations=false annotateNDA=false useNewAFCalculator=false heterozygosity=0.001 indel_heterozygosity=1.25E-4 heterozygosity_stdev=0.01 standard_min_confidence_threshold_for_calling=-0.0 standard_min_confidence_threshold_for_emitting=30.0 max_alternate_alleles=6 max_genotype_count=1024 max_num_PL_values=100 input_prior=[] sample_ploidy=2 genotyping_mode=DISCOVERY alleles=(RodBinding name= source=UNBOUND) contamination_fraction_to_filter=0.0 contamination_fraction_per_sample_file=null p_nonref_model=null exactcallslog=null output_mode=EMIT_VARIANTS_ONLY allSitePLs=true gcpHMM=10 pair_hmm_implementation=FASTEST_AVAILABLE phredScaledGlobalReadMismappingRate=45 noFpga=false nativePairHmmThreads=1 useDoublePrecision=false sample_name=null kmerSize=[10, 25] dontIncreaseKmerSizesForCycles=false allowNonUniqueKmersInRef=false numPruningSamples=1 recoverDanglingHeads=false doNotRecoverDanglingBranches=false minDanglingBranchLength=4 consensus=false maxNumHaplotypesInPopulation=128 errorCorrectKmers=false minPruning=2 debugGraphTransformations=false allowCyclesInKmerGraphToGeneratePaths=false graphOutput=null kmerLengthForReadErrorCorrection=25 minObservationsForKmerToBeSolid=20 GVCFGQBands=[5, 10, 15, 20, 30, 40, 50, 60] indelSizeToEliminateInRefModel=10 min_base_quality_score=10 includeUmappedReads=false useAllelesTrigger=false doNotRunPhysicalPhasing=false keepRG=null justDetermineActiveRegions=false dontGenotype=false dontUseSoftClippedBases=false captureAssemblyFailureBAM=false errorCorrectReads=false pcr_indel_model=CONSERVATIVE maxReadsInRegionPerSample=10000 minReadsPerAlignmentStart=10 mergeVariantsViaLD=false activityProfileOut=null activeRegionOut=null activeRegionIn=null activeRegionExtension=null forceActive=false activeRegionMaxSize=null bandPassSigma=null maxReadsInMemoryPerSample=30000 maxTotalReadsInMemory=10000000 maxProbPropagationDistance=50 activeProbabilityThreshold=0.002 min_mapping_quality_score=20 filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false\">\n",
      "\n",
      "STRUCTURED GATKCommandLine.HaplotypeCaller\n",
      "##GATKCommandLine.SelectVariants=<ID=SelectVariants,Version=3.8-0-ge9d806836,Date=\"Tue May 28 02:33:59 CEST 2019\",Epoch=1559003639148,CommandLineOptions=\"analysis_type=SelectVariants input_file=[] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=null excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/data/GENOMES/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=1000 baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=1 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false variant=(RodBinding name=variant source=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.raw_variants.vcf) discordance=(RodBinding name= source=UNBOUND) concordance=(RodBinding name= source=UNBOUND) out=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.raw_snps.vcf sample_name=[] sample_expressions=null sample_file=null exclude_sample_name=[] exclude_sample_file=[] exclude_sample_expressions=[] selectexpressions=[] invertselect=false excludeNonVariants=false excludeFiltered=false preserveAlleles=false removeUnusedAlternates=false restrictAllelesTo=ALL keepOriginalAC=false keepOriginalDP=false mendelianViolation=false invertMendelianViolation=false mendelianViolationQualThreshold=0.0 select_random_fraction=0.0 remove_fraction_genotypes=0.0 selectTypeToInclude=[SNP, NO_VARIATION] selectTypeToExclude=[] keepIDs=null excludeIDs=null fullyDecode=false justRead=false maxIndelSize=2147483647 minIndelSize=0 maxFilteredGenotypes=2147483647 minFilteredGenotypes=0 maxFractionFilteredGenotypes=1.0 minFractionFilteredGenotypes=0.0 maxNOCALLnumber=2147483647 maxNOCALLfraction=1.0 setFilteredGtToNocall=false ALLOW_NONOVERLAPPING_COMMAND_LINE_SAMPLES=false forceValidOutput=false filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false\">\n",
      "\n",
      "STRUCTURED GATKCommandLine.SelectVariants\n",
      "##GATKCommandLine.VariantFiltration=<ID=VariantFiltration,Version=3.8-0-ge9d806836,Date=\"Tue May 28 02:38:31 CEST 2019\",Epoch=1559003911372,CommandLineOptions=\"analysis_type=VariantFiltration input_file=[] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=[/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/.queue/scatterGather/GermlineFilter-2-sg/temp_01_of_16/scatter.intervals] excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/data/GENOMES/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=1000 baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=1 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false variant=(RodBinding name=variant source=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.raw_snps.vcf) mask=(RodBinding name= source=UNBOUND) out=/sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/tmp/.queue/scatterGather/GermlineFilter-2-sg/temp_01_of_16/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_snps.vcf filterExpression=[HaplotypeScore > 13.0, QD < 2.0, MQRankSum < -12.5, MQ < 40.0, ReadPosRankSum < -8.0, FS > 60.0] filterName=[SNP_HaplotypeScoreHigh, SNP_LowQualityDepth, SNP_MQRankSumLow, SNP_MappingQuality, SNP_ReadPosRankSumLow, SNP_StrandBias] genotypeFilterExpression=[] genotypeFilterName=[] clusterSize=3 clusterWindowSize=35 maskExtension=0 maskName=Mask filterNotInMask=false missingValuesInExpressionsShouldEvaluateAsFailing=false invalidatePreviousFilters=false invertFilterExpression=false invertGenotypeFilterExpression=false setFilteredGtToNocall=false filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false\">\n",
      "\n",
      "STRUCTURED GATKCommandLine.VariantFiltration\n",
      "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles in called genotypes\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=BaseQRankSum,Number=1,Type=Float,Description=\"Z-score from Wilcoxon rank sum test of Alt Vs. Ref base qualities\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=ClippingRankSum,Number=1,Type=Float,Description=\"Z-score From Wilcoxon rank sum test of Alt vs. Ref number of hard clipped bases\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=DB,Number=0,Type=Flag,Description=\"dbSNP Membership\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Approximate read depth; some reads may have been filtered\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=DS,Number=0,Type=Flag,Description=\"Were any of the samples downsampled?\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=END,Number=1,Type=Integer,Description=\"Stop position of the interval\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=ExcessHet,Number=1,Type=Float,Description=\"Phred-scaled p-value for exact test of excess heterozygosity\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=FS,Number=1,Type=Float,Description=\"Phred-scaled p-value using Fisher's exact test to detect strand bias\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=HaplotypeScore,Number=1,Type=Float,Description=\"Consistency of the site with at most two segregating haplotypes\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=InbreedingCoeff,Number=1,Type=Float,Description=\"Inbreeding coefficient as estimated from the genotype likelihoods per-sample when compared against the Hardy-Weinberg expectation\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=MQ,Number=1,Type=Float,Description=\"RMS Mapping Quality\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=MQRankSum,Number=1,Type=Float,Description=\"Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=QD,Number=1,Type=Float,Description=\"Variant Confidence/Quality by Depth\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=RAW_MQ,Number=1,Type=Float,Description=\"Raw data for RMS Mapping Quality\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=ReadPosRankSum,Number=1,Type=Float,Description=\"Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=SOR,Number=1,Type=Float,Description=\"Symmetric Odds Ratio of 2x2 contingency table to detect strand bias\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=set,Number=1,Type=String,Description=\"Source VCF for the merged record in CombineVariants\">\n",
      "\n",
      "INFO INFO\n",
      "##contig=<ID=1,length=249250621>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=2,length=243199373>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=3,length=198022430>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=4,length=191154276>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=5,length=180915260>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=6,length=171115067>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=7,length=159138663>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=8,length=146364022>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=9,length=141213431>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=10,length=135534747>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=11,length=135006516>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=12,length=133851895>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=13,length=115169878>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=14,length=107349540>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=15,length=102531392>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=16,length=90354753>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=17,length=81195210>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=18,length=78077248>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=19,length=59128983>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=20,length=63025520>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=21,length=48129895>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=22,length=51304566>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=X,length=155270560>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=Y,length=59373566>\n",
      "\n",
      "CONTIG contig\n",
      "##contig=<ID=MT,length=16569>\n",
      "\n",
      "CONTIG contig\n",
      "##reference=file:///data/GENOMES/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta\n",
      "\n",
      "GENERIC reference\n",
      "##source=SelectVariants\n",
      "\n",
      "GENERIC source\n",
      "##SnpEffVersion=\"4.3s (build 2017-10-25 10:05), by Pablo Cingolani\"\n",
      "\n",
      "GENERIC SnpEffVersion\n",
      "##SnpEffCmd=\"SnpEff  GRCh37.75 /sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_variants.vcf -lof -no-downstream -no-intergenic -noShiftHgvs \"\n",
      "\n",
      "GENERIC SnpEffCmd\n",
      "##INFO=<ID=ANN,Number=.,Type=String,Description=\"Functional annotations: 'Allele | Annotation | Annotation_Impact | Gene_Name | Gene_ID | Feature_Type | Feature_ID | Transcript_BioType | Rank | HGVS.c | HGVS.p | cDNA.pos / cDNA.length | CDS.pos / CDS.length | AA.pos / AA.length | Distance | ERRORS / WARNINGS / INFO'\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=LOF,Number=.,Type=String,Description=\"Predicted loss of function effects for this variant. Format: 'Gene_Name | Gene_ID | Number_of_transcripts_in_gene | Percent_of_transcripts_affected'\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=NMD,Number=.,Type=String,Description=\"Predicted nonsense mediated decay effects for this variant. Format: 'Gene_Name | Gene_ID | Number_of_transcripts_in_gene | Percent_of_transcripts_affected'\">\n",
      "\n",
      "INFO INFO\n",
      "##SnpSiftVersion=\"SnpSift 4.3s (build 2017-10-25 10:05), by Pablo Cingolani\"\n",
      "\n",
      "GENERIC SnpSiftVersion\n",
      "##SnpSiftCmd=\"\"\n",
      "\n",
      "GENERIC SnpSiftCmd\n",
      "##INFO=<ID=dbNSFP_phastCons46way_placental_rankscore,Number=A,Type=Float,Description=\"Field 'phastCons46way_placental_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_SAS_AF,Number=A,Type=Float,Description=\"Field 'ExAC_SAS_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_GERP___RS,Number=A,Type=Float,Description=\"Field 'GERP++_RS' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Ensembl_transcriptid,Number=A,Type=String,Description=\"Field 'Ensembl_transcriptid' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phastCons100way_vertebrate_rankscore,Number=A,Type=Float,Description=\"Field 'phastCons100way_vertebrate_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_SAS_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_SAS_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ARIC5606_AA_AF,Number=A,Type=Float,Description=\"Field 'ARIC5606_AA_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ARIC5606_AA_AC,Number=A,Type=Integer,Description=\"Field 'ARIC5606_AA_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MetaSVM_pred,Number=A,Type=Character,Description=\"Field 'MetaSVM_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_VEST3_score,Number=A,Type=Float,Description=\"Field 'VEST3_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_FATHMM_pred,Number=A,Type=Character,Description=\"Field 'FATHMM_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phyloP46way_placental,Number=A,Type=Float,Description=\"Field 'phyloP46way_placental' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MutationAssessor_rankscore,Number=A,Type=Float,Description=\"Field 'MutationAssessor_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ARIC5606_EA_AF,Number=A,Type=Float,Description=\"Field 'ARIC5606_EA_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SIFT_score,Number=A,Type=Float,Description=\"Field 'SIFT_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ARIC5606_EA_AC,Number=A,Type=Integer,Description=\"Field 'ARIC5606_EA_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_LRT_converted_rankscore,Number=A,Type=Float,Description=\"Field 'LRT_converted_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phyloP100way_vertebrate_rankscore,Number=A,Type=Float,Description=\"Field 'phyloP100way_vertebrate_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_AC,Number=A,Type=Integer,Description=\"Field '1000Gp1_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_AF,Number=A,Type=Float,Description=\"Field '1000Gp1_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_FIN_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_FIN_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_FIN_AF,Number=A,Type=Float,Description=\"Field 'ExAC_FIN_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_aapos_FATHMM,Number=A,Type=String,Description=\"Field 'aapos_FATHMM' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MetaSVM_rankscore,Number=A,Type=Float,Description=\"Field 'MetaSVM_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_FATHMM_score,Number=A,Type=Float,Description=\"Field 'FATHMM_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_cds_strand,Number=A,Type=Character,Description=\"Field 'cds_strand' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_LRT_score,Number=A,Type=Float,Description=\"Field 'LRT_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_EAS_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_EAS_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Ancestral_allele,Number=A,Type=Character,Description=\"Field 'Ancestral_allele' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_EAS_AF,Number=A,Type=Float,Description=\"Field 'ExAC_EAS_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ESP6500_EA_AF,Number=A,Type=Float,Description=\"Field 'ESP6500_EA_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_codonpos,Number=A,Type=Integer,Description=\"Field 'codonpos' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_Adj_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_Adj_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_Adj_AF,Number=A,Type=Float,Description=\"Field 'ExAC_Adj_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_fold_degenerate,Number=A,Type=Integer,Description=\"Field 'fold-degenerate' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_clinvar_rs,Number=A,Type=String,Description=\"Field 'clinvar_rs' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SIFT_converted_rankscore,Number=A,Type=Float,Description=\"Field 'SIFT_converted_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Ensembl_geneid,Number=A,Type=String,Description=\"Field 'Ensembl_geneid' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_hg38_pos,Number=A,Type=Integer,Description=\"Field 'hg38_pos' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Polyphen2_HDIV_rankscore,Number=A,Type=Float,Description=\"Field 'Polyphen2_HDIV_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MutationTaster_converted_rankscore,Number=A,Type=Float,Description=\"Field 'MutationTaster_converted_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phyloP100way_vertebrate,Number=A,Type=Float,Description=\"Field 'phyloP100way_vertebrate' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_LRT_pred,Number=A,Type=Character,Description=\"Field 'LRT_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_PROVEAN_pred,Number=A,Type=Character,Description=\"Field 'PROVEAN_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phastCons100way_vertebrate,Number=A,Type=Float,Description=\"Field 'phastCons100way_vertebrate' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_VEST3_rankscore,Number=A,Type=Float,Description=\"Field 'VEST3_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MetaSVM_score,Number=A,Type=Float,Description=\"Field 'MetaSVM_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_CADD_phred,Number=A,Type=Float,Description=\"Field 'CADD_phred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_FATHMM_rankscore,Number=A,Type=Float,Description=\"Field 'FATHMM_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_refcodon,Number=A,Type=String,Description=\"Field 'refcodon' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_aapos_SIFT,Number=A,Type=String,Description=\"Field 'aapos_SIFT' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_clinvar_trait,Number=A,Type=String,Description=\"Field 'clinvar_trait' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ESP6500_AA_AF,Number=A,Type=Float,Description=\"Field 'ESP6500_AA_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SLR_test_statistic,Number=A,Type=Float,Description=\"Field 'SLR_test_statistic' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_PROVEAN_score,Number=A,Type=Float,Description=\"Field 'PROVEAN_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SiPhy_29way_logOdds,Number=A,Type=Float,Description=\"Field 'SiPhy_29way_logOdds' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_NFE_AF,Number=A,Type=Float,Description=\"Field 'ExAC_NFE_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_GERP___NR,Number=A,Type=Float,Description=\"Field 'GERP++_NR' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_UniSNP_ids,Number=A,Type=String,Description=\"Field 'UniSNP_ids' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Polyphen2_HVAR_score,Number=A,Type=Float,Description=\"Field 'Polyphen2_HVAR_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phastCons46way_primate,Number=A,Type=Float,Description=\"Field 'phastCons46way_primate' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_AMR_AF,Number=A,Type=Float,Description=\"Field '1000Gp1_AMR_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_AMR_AC,Number=A,Type=Integer,Description=\"Field '1000Gp1_AMR_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_PROVEAN_converted_rankscore,Number=A,Type=Float,Description=\"Field 'PROVEAN_converted_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Uniprot_id,Number=A,Type=String,Description=\"Field 'Uniprot_id' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Interpro_domain,Number=A,Type=String,Description=\"Field 'Interpro_domain' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phastCons46way_placental,Number=A,Type=Float,Description=\"Field 'phastCons46way_placental' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SiPhy_29way_logOdds_rankscore,Number=A,Type=Float,Description=\"Field 'SiPhy_29way_logOdds_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Reliability_index,Number=A,Type=Integer,Description=\"Field 'Reliability_index' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_AFR_AF,Number=A,Type=Float,Description=\"Field 'ExAC_AFR_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_CADD_raw,Number=A,Type=Float,Description=\"Field 'CADD_raw' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_AFR_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_AFR_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Polyphen2_HVAR_rankscore,Number=A,Type=Float,Description=\"Field 'Polyphen2_HVAR_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_GERP___RS_rankscore,Number=A,Type=Float,Description=\"Field 'GERP++_RS_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phyloP46way_primate_rankscore,Number=A,Type=Float,Description=\"Field 'phyloP46way_primate_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MetaLR_score,Number=A,Type=Float,Description=\"Field 'MetaLR_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Polyphen2_HDIV_score,Number=A,Type=Float,Description=\"Field 'Polyphen2_HDIV_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_aapos,Number=A,Type=Integer,Description=\"Field 'aapos' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_clinvar_clnsig,Number=A,Type=Integer,Description=\"Field 'clinvar_clnsig' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_AFR_AC,Number=A,Type=Integer,Description=\"Field '1000Gp1_AFR_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_AFR_AF,Number=A,Type=Float,Description=\"Field '1000Gp1_AFR_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MetaLR_rankscore,Number=A,Type=Float,Description=\"Field 'MetaLR_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_AMR_AF,Number=A,Type=Float,Description=\"Field 'ExAC_AMR_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MutationTaster_pred,Number=A,Type=Character,Description=\"Field 'MutationTaster_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MetaLR_pred,Number=A,Type=Character,Description=\"Field 'MetaLR_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_AMR_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_AMR_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_NFE_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_NFE_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_CADD_raw_rankscore,Number=A,Type=Float,Description=\"Field 'CADD_raw_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_genename,Number=A,Type=String,Description=\"Field 'genename' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MutationAssessor_score,Number=A,Type=Float,Description=\"Field 'MutationAssessor_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_COSMIC_ID,Number=A,Type=String,Description=\"Field 'COSMIC_ID' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phastCons46way_primate_rankscore,Number=A,Type=Float,Description=\"Field 'phastCons46way_primate_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SiPhy_29way_pi,Number=A,Type=String,Description=\"Field 'SiPhy_29way_pi' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_AF,Number=A,Type=Float,Description=\"Field 'ExAC_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Uniprot_acc,Number=A,Type=String,Description=\"Field 'Uniprot_acc' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_ExAC_AC,Number=A,Type=Integer,Description=\"Field 'ExAC_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_COSMIC_CNT,Number=A,Type=Integer,Description=\"Field 'COSMIC_CNT' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phyloP46way_placental_rankscore,Number=A,Type=Float,Description=\"Field 'phyloP46way_placental_rankscore' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Polyphen2_HDIV_pred,Number=A,Type=Character,Description=\"Field 'Polyphen2_HDIV_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_ASN_AF,Number=A,Type=Float,Description=\"Field '1000Gp1_ASN_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_ASN_AC,Number=A,Type=Integer,Description=\"Field '1000Gp1_ASN_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_LRT_Omega,Number=A,Type=Float,Description=\"Field 'LRT_Omega' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_EUR_AC,Number=A,Type=Integer,Description=\"Field '1000Gp1_EUR_AC' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_hg38_chr,Number=A,Type=String,Description=\"Field 'hg38_chr' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MutationTaster_score,Number=A,Type=Float,Description=\"Field 'MutationTaster_score' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Uniprot_aapos,Number=A,Type=Integer,Description=\"Field 'Uniprot_aapos' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_MutationAssessor_pred,Number=A,Type=Character,Description=\"Field 'MutationAssessor_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_1000Gp1_EUR_AF,Number=A,Type=Float,Description=\"Field '1000Gp1_EUR_AF' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_Polyphen2_HVAR_pred,Number=A,Type=Character,Description=\"Field 'Polyphen2_HVAR_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_SIFT_pred,Number=A,Type=Character,Description=\"Field 'SIFT_pred' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=dbNSFP_phyloP46way_primate,Number=A,Type=Float,Description=\"Field 'phyloP46way_primate' from dbNSFP\">\n",
      "\n",
      "INFO INFO\n",
      "##bcftools_annotateVersion=1.3.1+htslib-1.3.1\n",
      "\n",
      "GENERIC bcftools_annotateVersion\n",
      "##bcftools_annotateCommand=annotate -a /data/common_dbs/cosmic_v85/CosmicCodingMuts_v85_collapsed.vcf.gz -c =ID -o /sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_variants_snpEff_snpSift_Cosmicv85.vcf -O v /sample/output/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079/190524_HMFregCPCT_FR17036004_FR17436860_CPCT02220079.filtered_variants_snpEff_snpSift.vcf.gz\n",
      "\n",
      "GENERIC bcftools_annotateCommand\n",
      "##INFO=<ID=GoNLv5_AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=GoNLv5_AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "\n",
      "INFO INFO\n",
      "##INFO=<ID=GoNLv5_AN,Number=1,Type=Integer,Description=\"Total number of alleles in called genotypes\">\n",
      "\n",
      "INFO INFO\n"
     ]
    }
   ],
   "source": [
    ">>> for x in vcf_in.header.records:\n",
    ">>>    print (x)\n",
    ">>>    print (x.type, x.key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f060bd55",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
